servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/ScanWithMapperExt.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api;
+
+import io.servicetalk.concurrent.PublisherSource.Subscriber;
+
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+/**
+ * Provides the ability to transform (aka map) signals emitted via the {@link Publisher#scanWithMapper(Supplier)} operator.
+ * @param <T> Type of items emitted by the {@link Publisher} this operator is applied.
+ * @param <R> Type of items emitted by this operator.
+ */
+public interface ScanWithMapperExt<T, R> {
```

Comment:
Naming suggestions are welcome. "Ext" was meant to be short hand for "Extended" and follows similar pattern as `X509ExtendedTrustManager` and `X509TrustManager`